lib/prometheus-parser/src/lib.rs

```diff
@@ -1,12 +1,34 @@
+use indexmap::IndexMap;
 use snafu::ResultExt;
-use std::collections::BTreeMap;
+use std::{collections::BTreeMap, convert::TryFrom};
 
 mod line;
 
 pub use line::ErrorKind;
-use line::Line;
-use line::Metric;
-use line::MetricKind;
+use line::{Line, Metric, MetricKind};
+
+pub const METRIC_NAME_LABEL: &str = "__name__";
+
+pub mod proto {
+    include!(concat!(env!("OUT_DIR"), "/prometheus.rs"));
+
+    pub use metric_metadata::MetricType;
+
+    impl MetricType {
+        pub fn as_str(&self) -> &'static str {
+            match self {
+                MetricType::Counter => "counter",
+                MetricType::Gauge => "gauge",
+                MetricType::Histogram => "histogram",
+                MetricType::Summary => "summary",
+                MetricType::Gaugehistogram => "gaugehistogram",
```

Comment:
```suggestion
                MetricType::GaugeHistogram => "gauge_histogram",
```

(Not sure in which context the string is used, `gauge histogram` or `gaugehistogram` might be appropriate if that's either a human description or magic string.)

Reply:
It is a semi-magic string, used in output generated by the `prometheus_exporter` sink. It appears in a `# TYPE` comment line indicating the type of a particular metric set.